prow/jobs/kyma-project/telemetry-manager/telemetry-manager-upgrade-test.yaml

```diff
@@ -0,0 +1,39 @@
+presubmits:
+  kyma-project/telemetry-manager:
+    - name: pull-telemetry-manager-upgrade-test
+      annotations:
+        description: "Pre-submit job to execute upgrade test."
+        owner: "huskies"
+      optional: false
+      labels:
+        preset-dind-enabled: "true"
+        preset-kind-volume-mounts: "true"
+        prow.k8s.io/pubsub.project: "sap-kyma-prow"
+        prow.k8s.io/pubsub.runID: "pull-telemetry-manager-upgrade-test"
+        prow.k8s.io/pubsub.topic: "prowjobs"
+      cluster: untrusted-workload
+      decorate: true
+      run_if_changed: '^(go.mod|go.sum)$|.*\.go$|Dockerfile|Makefile|^config/'
+      spec:
+        containers:
+          - image: europe-docker.pkg.dev/kyma-project/prod/testimages/e2e-dind-k3d:v20230612-dadc01a7
+            command: ["/init.sh"]
+            args: ["bash", "-c", "make upgrade-test"]
+            requests:
+              memory: 4Gi
+              cpu: 2
+            limits:
+              memory: 4Gi
+              cpu: 2
+            securityContext:
+              privileged: true
+              seccompProfile:
+                type: Unconfined
+              allowPrivilegeEscalation: true
+        tolerations:
+          - key: dedicated
+            operator: Equal
+            value: high-cpu
+            effect: NoSchedule
+        nodeSelector:
+          dedicated: "high-cpu"
```

Comment:
what's the impact in time of execution if you remove the toleration and schedule the job in standard pool?

Reply:
I didn't try to execute it without the toleration as I wanted a similar setup to the one we use for our e2e-tests https://github.com/kyma-project/test-infra/blob/main/prow/jobs/kyma-project/telemetry-manager/telemetry-manager-e2e-test.yaml#L33 since the upgrade job will execute a subset of the e2e tests before upgrade and after upgrade

Reply:
I checked it out.

When we use a high-cpu the job executes in 3m 46s https://status.build.kyma-project.io/view/gs/kyma-prow-logs/pr-logs/pull/kyma-project_telemetry-manager/233/shorim_test_of_prowjob_pull-telemetry-manager-upgrade-test/1672151807964483584, while in the standard pool the job executes in 8m 59s https://status.build.kyma-project.io/view/gs/kyma-prow-logs/pr-logs/pull/kyma-project_telemetry-manager/233/shorim_test_of_prowjob_pull-telemetry-manager-upgrade-test/1672258710405124096

~9m is still an acceptable time for us, so I removed the dedication to high-cpu